src/app/EventLogging.h

```diff
@@ -53,15 +55,36 @@ class EventLogger : public EventLoggingDelegate
  * context tags to be interpreted within the schema identified by
  * `ClusterID` and `EventId`.
  *
- * @param[in] apDelegate The EventLoggingDelegate to serialize the event data
+ * LogEvent has 2 variant, for aEventData that have method GetFabricIndex with return type FabricIndex,
+ * it would use first  logEvent function which can process the fabric index, for
+ * aEventData that don't have GetFabricIndex, it would use second LogEvent function.
+ * @param[in] aEventData  The event cluster object
  * @param[in] aEndpoint    The current cluster's Endpoint Id
  * @param[in] aUrgent    The EventOption Type, kUrgent or kNotUrgent
  * @param[out] aEventNumber The event Number if the event was written to the
  *                         log, 0 otherwise. The Event number is expected to monotonically increase.
  *
  * @return CHIP_ERROR  CHIP Error Code
  */
-template <typename T>
+template <typename T, std::enable_if_t<DataModel::IsFabricScoped<T>::value, bool> = true>
+CHIP_ERROR LogEvent(const T & aEventData, EndpointId aEndpoint, EventNumber & aEventNumber,
+                    EventOptions::Type aUrgent = EventOptions::Type::kNotUrgent)
+{
+    EventLogger<T> eventData(aEventData);
+    ConcreteEventPath path(aEndpoint, aEventData.GetClusterId(), aEventData.GetEventId());
+    EventManagement & logMgmt = chip::app::EventManagement::GetInstance();
+    EventOptions eventOptions;
+    eventOptions.mUrgent   = aUrgent;
+    eventOptions.mPath     = path;
+    eventOptions.mPriority = aEventData.GetPriorityLevel();
+    if (aEventData.GetFabricIndex() != kUndefinedFabricIndex)
```

Comment:
Why?  What cases is this trying to handle, and why is handling them by treating those events as not fabric scoped the right thing?

Reply:
right, just remove this check, and also remove optional fabricindex in this PR, I think it is fine to use 0, kUndefinedFabricIndex for "no fabric"